File: examples/src/main/java/org/apache/ignite/examples/model/Organization.java
/*
 * Copyright 2019 GridGain Systems, Inc. and Contributors.
 *
 * Licensed under the GridGain Community Edition License (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.ignite.examples.model;

import java.sql.Timestamp;
import java.util.concurrent.atomic.AtomicLong;
import org.apache.ignite.cache.query.annotations.QuerySqlField;

/**
 * This class represents organization object.
 */
public class Organization {
    /** */
    private static final AtomicLong ID_GEN = new AtomicLong();

    /** Organization ID (indexed). */
    @QuerySqlField(index = true)
    private Long id;

    /** Organization name (indexed). */
    @QuerySqlField(index = true)
    private String name;

    /** Address. */
    private Address addr;

    /** Type. */
    private OrganizationType type;

    /** Last update time. */
    private Timestamp lastUpdated;

    /**
     * Required for binary deserialization.
     */
    public Organization() {
        // No-op.
    }

    /**
     * @param name Organization name.
     */
    public Organization(String name) {
        id = ID_GEN.incrementAndGet();

        this.name = name;
    }

    /**
     * @param id Organization ID.
     * @param name Organization name.
     */
    public Organization(long id, String name) {
        this.id = id;
        this.name = name;
    }

    /**
     * @param name Name.
     * @param addr Address.
     * @param type Type.
     * @param lastUpdated Last update time.
     */
    public Organization(String name, Address addr, OrganizationType type, Timestamp lastUpdated) {
        id = ID_GEN.incrementAndGet();

        this.name = name;
        this.addr = addr;
        this.type = type;

        this.lastUpdated = lastUpdated;
    }

    /**
     * @return Organization ID.
     */
    public Long id() {
        return id;
    }

    /**
     * @return Name.
     */
    public String name() {
        return name;
    }

    /**
     * @return Address.
     */
    public Address address() {
        return addr;
    }

    /**
     * @return Type.
     */
    public OrganizationType type() {
        return type;
    }

    /**
     * @return Last update time.
     */
    public Timestamp lastUpdated() {
        return lastUpdated;
    }

    /** {@inheritDoc} */
    @Override public String toString() {
        return "Organization [id=" + id +
            ", name=" + name +
            ", address=" + addr +
            ", type=" + type +
            ", lastUpdated=" + lastUpdated + ']';
    }
}

File: examples/src/main/scala/org/apache/ignite/scalar/examples/model/Organization.scala
/*
 * ___    _________________________ ________
 * __ |  / /____  _/__  ___/__  __ \___  __ \
 * __ | / /  __  /  _____ \ _  / / /__  /_/ /
 * __ |/ /  __/ /   ____/ / / /_/ / _  _, _/
 * _____/   /___/   /____/  \____/  /_/ |_|
 *
 */

package org.apache.ignite.scalar.examples.model

import java.sql.Timestamp
import java.util.concurrent.atomic.AtomicLong

import org.apache.ignite.cache.query.annotations.QuerySqlField
import org.apache.ignite.examples.model.{Address, OrganizationType}

class Organization {
    /** */
    private[this] val ID_GEN = new AtomicLong

    /** Organization ID (indexed). */
    @QuerySqlField(index = true) private[this] var id = 0L

    /** Organization name (indexed). */
    @QuerySqlField(index = true) private[this] var name: String = null

    /** Address. */
    private[this] var addr: Address = null

    /** Type. */
    private[this] var orgType: OrganizationType = null

    /** Last update time. */
    private[this] var lastUpdated: Timestamp = null

    /**
      * @param name Organization name.
      */
    def this(name: String) {
        this()

        id = ID_GEN.incrementAndGet
        this.name = name
    }

    /**
      * @param name Name.
      * @param addr Address.
      * @param type Type.
      * @param lastUpdated Last update time.
      */
    def this(name: String, addr: Address, `type`: OrganizationType, lastUpdated: Timestamp) {
        this()

        id = ID_GEN.incrementAndGet
        this.name = name
        this.addr = addr
        this.orgType = `type`
        this.lastUpdated = lastUpdated
    }

    /**
      * @return Organization ID.
      */
    def getId: Long = {
        id
    }

    /**
      * @return Name.
      */
    def getName: String = {
        name
    }

    /**
      * @return Address.
      */
    def getAddress: Address = {
        addr
    }

    /**
      * @return Type.
      */
    def getOrganizationType: OrganizationType = {
        orgType
    }

    /**
      * @return Last update time.
      */
    def getLastUpdated: Timestamp = {
        lastUpdated
    }

    /** @inheritdoc */
    override def toString: String = {
        "Organization [id=" + id +
            ", name=" + name +
            ", address=" + addr +
            ", type=" + orgType +
            ", lastUpdated=" + lastUpdated +
            ']'
    }
}
